src/v/kafka/server/group.h

```diff
@@ -587,6 +587,16 @@ class group {
         const group& _group;
     };
 
+    ss::lw_shared_ptr<mutex> get_tx_lock(model::producer_id pid) {
+        auto lock_it = _tx_locks.find(pid);
+        if (lock_it == _tx_locks.end()) {
+            auto [new_it, _] = _tx_locks.try_emplace(
+              pid, ss::make_lw_shared<mutex>());
+            lock_it = new_it;
+        }
+        return lock_it->second;
```

Comment:
I think this entire thing is equivalent to `return _tx_locks[pid]` which will default construct the mutex if it doesn't already exist (and avoids the double lookup with try_emplace)

Reply:
done

Reply:
A default constructed `lw_shared_ptr` is `nullptr`. But can the container be `absl::flat_hash_map<model::producer_id, mutex>`?

Reply:
nice catch, I've changed the container